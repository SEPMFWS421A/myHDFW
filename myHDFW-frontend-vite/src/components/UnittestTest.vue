<template>
    <Card style="width: 35rem; overflow: hidden; margin: auto;">
        <template #header>
        <RouterLink to="/Schedule/">
            <img id="nav_card_image" class="nav_card_image" alt="user header" src="@/assets/Eventverwaltung.png" />
        </RouterLink>
        </template>
        <template #title>
        <span class="nav_card_title">Eventverwaltung</span>
        <Button id="help_button_unittestTest" class="help_button" label="" icon="pi pi-question-circle" v-tooltip="{ value: 'Confirm to proceed', showDelay: 1000, hideDelay: 300 }"/>
        </template>
        <template #content>
        <Toolbar>
            <template #start>
            <Button id="add_unittestTest" class="add_unittestTest" data-pc-severity="none" label="Event hinzufügen" icon="pi pi-plus" severity="success"  @click.native="openNew" />
            </template>

            <template #end>
            <Button id="delete_unittestTest" class="delete_unittestTest" label="Event löschen" icon="pi pi-trash" @click="confirmDeleteSelected" :disabled="isDisabled" />
            </template>
        </Toolbar>
        </template>
    </Card>
</template>

<script setup lang="js">
import {onMounted, ref} from 'vue';

function openNew(params) {
    
}

function confirmDeleteSelected(params) {
    
}

function isDisabled(params) {
    return true;
}
</script>