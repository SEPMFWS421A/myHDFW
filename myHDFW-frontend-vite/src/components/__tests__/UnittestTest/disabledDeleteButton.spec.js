/*
import { mount } from '@vue/test-utils'
import UnittestTest from '@/components/UnittestTest.vue'
import { describe, test, expect } from 'vitest'

describe('UnittestTest', () => {
  test('renders correctly', () => {
    const wrapper = mount(UnittestTest)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('clicking "add_unittestTest" button calls "openNew" method', async () => {
    const openNewMock = jest.fn()
    const wrapper = mount(UnittestTest, {
      methods: {
        openNew: openNewMock
      }
    })
    await wrapper.find('#add_unittestTest').trigger('click')
    expect(openNewMock).toHaveBeenCalled()
  })

  test('clicking "delete_unittestTest" button calls "confirmDeleteSelected" method', async () => {
    const confirmDeleteSelectedMock = jest.fn()
    const wrapper = mount(UnittestTest, {
      methods: {
        confirmDeleteSelected: confirmDeleteSelectedMock
      }
    })
    await wrapper.find('#delete_unittestTest').trigger('click')
    expect(confirmDeleteSelectedMock).toHaveBeenCalled()
  })

  test('"delete_unittestTest" button is disabled', () => {
    const wrapper = mount(UnittestTest)
    expect(wrapper.find('#delete_unittestTest').element.disabled).toBe(true)
  })
})
*/
import { mount } from 'vue-test-utils'
import { nextTick } from 'vue'
import UnittestTest from '@/components/UnittestTest.vue'
import { describe, test, expect } from 'vitest'


describe('UnittestTest', () => {
  it('renders correctly', () => {
    const wrapper = mount(UnittestTest)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('clicking "add_unittestTest" button calls "openNew" method', async () => {
    const openNewMock = jest.fn()
    const wrapper = mount(UnittestTest, {
      methods: {
        openNew: openNewMock
      }
    })
    await wrapper.find('#add_unittestTest').trigger('click')
    await nextTick()
    expect(openNewMock).toHaveBeenCalled()
  })

  it('clicking "delete_unittestTest" button calls "confirmDeleteSelected" method', async () => {
    const confirmDeleteSelectedMock = jest.fn()
    const wrapper = mount(UnittestTest, {
      methods: {
        confirmDeleteSelected: confirmDeleteSelectedMock
      }
    })
    await wrapper.find('#delete_unittestTest').trigger('click')
    await nextTick()
    expect(confirmDeleteSelectedMock).toHaveBeenCalled()
  })

  it('"delete_unittestTest" button is disabled', () => {
    const wrapper = mount(UnittestTest)
    expect(wrapper.find('#delete_unittestTest').attributes().disabled).toBe('disabled')
  })
})
