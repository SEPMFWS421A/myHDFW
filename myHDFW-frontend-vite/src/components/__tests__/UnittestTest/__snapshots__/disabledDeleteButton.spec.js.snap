// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UnittestTest > renders correctly 1`] = `
<card
  style="width: 35rem; overflow: hidden; margin: auto;"
/>
`;
